lib/flight_job/models/script.rb

```diff
@@ -213,6 +211,23 @@ def metadata_path
       @metadata_path ||= File.join(FlightJob.config.scripts_dir, id, 'metadata.yaml')
     end
 
+    def initialize_notes(notes)
+      @notes = Notes.new(id, notes)
+    end
+
+    def notes
+      @notes ||= if File.exist?(notes_path)
+                   Notes.new(id, Notes.load_from_path(notes_path))
+                 else
+                   Notes.new(id)
+                 end
```

Comment:
Class methods called `load*` ought to be returning an instance of the class, also the branching logic here seems like it would better live in a class method on `Note`.  I think this would read much better if it was

```ruby
def notes
  @notes ||= Notes.load(id)
end
```

`Notes::load` would then be responsible for constructing the path, instead of both `Note` and `Script` currently doing so.